domain/src/main/java/org/oppia/domain/profile/ProfileManagementController.kt

```diff
@@ -98,15 +373,69 @@ class ProfileManagementController @Inject constructor(
     return ProfileId.newBuilder().setInternalId(currentProfileId).build()
   }
 
-  /**
-   * Sets the currentProfileId to the selected profile in ProfileChooserFragment.
-   * Checks to ensure that profileId corresponds to an existing profile.
-   *
-   * @param profileId the ID corresponding to the profile being set.
-   * @return a [LiveData] that indicates the success/failure of this set operation.
-   */
-  fun setCurrentProfileId(profileId: ProfileId): LiveData<AsyncResult<Any?>> {
-    currentProfileId = profileId.internalId
-    return MutableLiveData(AsyncResult.success<Any?>(null))
+  private fun isUniqueName(newName: String, profileDatabase: ProfileDatabase): Boolean {
+    val lowerCaseNewName = newName.toLowerCase()
+    profileDatabase.profilesMap.values.forEach {
+      if (it.name.toLowerCase() == lowerCaseNewName) {
+        return false
+      }
+    }
+    return true
+  }
+
+  private fun saveImageToInternalStorage(avatarImagePath: Uri, profileDir: File): String {
+    val bitmap = MediaStore.Images.Media.getBitmap(context.contentResolver, avatarImagePath)
+    val imageFile = File(profileDir, "profile.png")
+    var fos: FileOutputStream? = null
+    try {
+      fos = FileOutputStream(imageFile)
+      rotateAndCompressBitmap(avatarImagePath, bitmap, 200)
+        .compress(Bitmap.CompressFormat.PNG, /** quality= */ 100, fos)
+    } catch (e: Exception) {
+      logger.e("ProfileManagementController", "Failed to store user submitted avatar image", e)
+      return ""
+    } finally {
+      try {
+        fos?.close()
+      } catch (e: IOException) {
+        logger.e("ProfileManagementController", "Failed to close FileOutputStream for avatar image", e)
+      }
+    }
+    return imageFile.absolutePath
+  }
+
+  private fun onlyLetters(name: String): Boolean {
+    return name.matches(Regex("^[ A-Za-z]+\$"))
+  }
+
+  // https://stackoverflow.com/questions/3934331/how-to-hash-a-string-in-android
+  private fun md5(s: String): String {
+    var digest: MessageDigest? = null
+    try {
+      digest = MessageDigest.getInstance("MD5")
+      digest.update(s.toByteArray(Charset.forName("US-ASCII")), 0, s.length)
+      val magnitude = digest.digest()
+      val bi = BigInteger(1, magnitude)
+      val hash = String.format("%0" + (magnitude.size.shl(1)) + "x", bi)
+      return hash
+    } catch (e: NoSuchAlgorithmException) {
+      logger.e("ProfileManagementController", "No such algorithm when creating md5 hash for gravatar", e)
+    }
+    return ""
+  }
+
+  private fun rotateAndCompressBitmap(uri: Uri, bitmap: Bitmap, cropSize: Int): Bitmap {
+    val croppedBitmap = ThumbnailUtils.extractThumbnail(bitmap, cropSize, cropSize)
+    val inputStream = context.contentResolver.openInputStream(uri)
+    val orientation = ExifInterface(inputStream).getAttributeInt(ExifInterface.TAG_ORIENTATION, 1)
```

Comment:
This is need to detect if the orientation of the image needs to be rotated. However, ExifInterface(inputStream) require API level 24, but this was the only way I found to successfully create the ExifInterface. I tried using the uri to get the absolute file path but that resulted in FileNotFoundException.